Solutions/Ais.Net.Receiver/StorageClient.cs

```diff
@@ -15,33 +16,58 @@ namespace Endjin.Ais.Receiver
     using System.Text;
     using System.Threading.Tasks;
 
-    public class StorageClient
+    public class StorageClient : IStorageClient
     {
-        private AppendBlobClient container;
-        private IConfiguration configuration;
-        private Stream stream;
+        private readonly IConfiguration configuration;
+        private AppendBlobClient appendBlobClient;
+        private BlobContainerClient blobContainerClient;
 
         public StorageClient(IConfiguration configuration)
         {
             this.configuration = configuration;
         }
 
-        public async Task AppendMessages(IEnumerable<string> messages)
+        public async Task PersistAsync(IEnumerable<string> messages)
         {
-            await this.InitialiseConnectionAsync().ConfigureAwait(false);
+            await this.InitialiseContainerAsync().ConfigureAwait(false);
+            await using MemoryStream stream = new MemoryStream(Encoding.UTF8.GetBytes(messages.Aggregate(new StringBuilder(), (sb, a) => sb.AppendLine(string.Join(",", a)), sb => sb.ToString())));
+            await this.appendBlobClient.AppendBlockAsync(stream).ConfigureAwait(false);
+        }
 
-            using (var stream = new MemoryStream(Encoding.UTF8.GetBytes(messages.Aggregate(new StringBuilder(), (sb, a) => sb.AppendLine(String.Join(",", a)), sb => sb.ToString())))) 
+        private async Task InitialiseContainerAsync()
+        {
+            try
+            {
+                this.blobContainerClient = new BlobContainerClient(
+                    this.configuration[ConfigurationKeys.ConnectionString],
+                    this.configuration[ConfigurationKeys.ContainerName]);
+                this.appendBlobClient = new AppendBlobClient(
+                    this.configuration[ConfigurationKeys.ConnectionString],
+                    this.configuration[ConfigurationKeys.ContainerName],
+                    $"raw/{DateTimeOffset.Now:yyyy}/{DateTimeOffset.Now:MM}/{DateTimeOffset.Now:dd}/{DateTimeOffset.Now:yyyyMMddTHH}.nm4");
```

Comment:
Three problems here.

1. this partitions data by when we happened to write it out to blob storage, and now by when we received it (possibly not a big problem)
2. You should use `DateTimeOffset.UtcNow` because `Now` will reflect daylight saving, and since you're not writing the timezone out you'll encounter the "this hour happened twice" problem when the clocks go back.
3. `DateTimeOffset.Now` is a moving target, so it's possible that the filename won't match the folder structure

For 2, e.g. suppose it's 2021/02/28 23:59:59.99999 at the point where you start running this. So your expression here:

```cs
$"raw/{DateTimeOffset.Now:yyyy}/{DateTimeOffset.Now:MM}/{DateTimeOffset.Now:dd}/{DateTimeOffset.Now:yyyyMMddTHH}.nm4"
```

starts by reading the `DateTimeNow` and gets a year of 2021, then it reads it again and gets a month of 02, but then before it manages to read it for the third time, the clock rolls past midnight, so now you get a day of 01, because it's March now. So you end up with this sort of filename:

`raw/2021/02/01/2021030100.nm4`

For this sort of thing you always want to read `DateTime.UtcNow` once, store the value in a local, and then refer to that local in all the places where you want to pull out the component parts.

The bug you have here is particularly infuriating because you will _almost_ never encounter it in reality, making it incredibly baffling when you finally do. (And as my father learned from spending half his career in real-time systems, timing bugs will always hit you eventually. It's not good enough to make them very unlikely: that just makes them harder to repro.)